MSOE.MediaComplete/Settings.xaml.cs

```diff
@@ -100,6 +101,11 @@ private void BtnSelectHomeFolder_Click(object sender, RoutedEventArgs e)
 
             if (!_allDirs.Contains(SettingWrapper.HomeDir.FullPath))
             {
+                var withParent = _allDirs.Where(x => x.StartsWith(SettingWrapper.HomeDir.FullPath, StringComparison.Ordinal));
+                foreach (var parent in withParent)
+                {
+                    _allDirs.Remove(parent);
+                }
```

Comment:
I genuinely have no idea what this code is doing. Is it canceling past home dirs that are just subdirectories of the new one? Why?